kinto/plugins/history/views.py

```diff
@@ -57,3 +61,83 @@ def _extract_filters(self):
             filters_str_id.append(filt)
 
         return filters_str_id
+
+
+version_view = Service(name='version_view',
+                       description='Handle retrieving object from the past.',
+                       path='/buckets/{bucket_id}{subpath:.*}/version/{last_modified:[0-9]{13}}')
+
+
+@version_view.get(permission=NO_PERMISSION_REQUIRED)
+def get_version_view(request):
+    last_modified = int(request.matchdict['last_modified'])
+    bucket_id = request.matchdict['bucket_id']
+    resource_path = '/buckets/{}{}'.format(request.matchdict['bucket_id'],
+                                           request.matchdict['subpath'])
+
+    bucket_uri = '/buckets/{}'.format(bucket_id)
+
+    is_collection = any([resource_path.endswith('/{}'.format(collection_type))
+                         for collection_type in ['collections', 'groups', 'records']])
+
+    # Handle collections
+    if is_collection:
+        return handle_version_on_collections(request, last_modified, bucket_uri)
+
+    # Handle records
+    return handle_version_on_records(request, last_modified, bucket_uri)
+
+
+def get_resource_name(request):
+    subpath = '/{}'.format(request.matchdict['subpath'])
+    resources = {
+        '/collections': "collection",
+        '/groups': 'group',
+        '/records': 'record'
+    }
+    for suffix, name in resources.items():
+        if subpath.endswith(suffix):
+            return name
+
+
+def handle_version_on_collections(request, last_modified, bucket_uri):
+    parent_id = '{}{}/*'.format(bucket_uri, request.matchdict['subpath'])
+    resource_name = get_resource_name(request)
+
+    # We want to get the record at a certain time.
+    filters = [Filter('target.data.last_modified', last_modified, COMPARISON.MAX),
+               Filter('resource_name', resource_name, COMPARISON.EQ),
+               Filter('uri', parent_id, COMPARISON.LIKE)]
+    sorting = [Sort('last_modified', -1)]
+
+    records, count = request.registry.storage.get_all(
+        collection_id='history', parent_id=bucket_uri,
+        filters=filters, sorting=sorting)
+
+    # Only process records once
+    already_processed = set()
+    results = []
+    for record in records:
+        record_id = record['target']['data']['id']
+        if record_id not in already_processed:
+            already_processed.add(record_id)
+            if record['target']['data'].get('deleted') is not True:
```

Comment:
I can see some use cases where deleted is a property of a record and not a tombstones: `"deleted": 3`